symengine/flint_wrapper.h

```diff
@@ -627,6 +667,70 @@ class fmpz_poly_wrapper
         return fmpz_poly_divrem(*q.get_fmpz_poly_t(), *r.get_fmpz_poly_t(),
                                 *get_fmpz_poly_t(), *b.get_fmpz_poly_t());
     }
+    static void fmpz_poly_factor(const fmpz_poly_t G, fmpz_poly_factor_t fac)
+    {
+        const slong lenG = G->length;
+        fmpz_poly_t g;
+
+        if (lenG == 0) {
+            fmpz_set_ui(&fac->c, 0);
+            return;
+        }
+        if (lenG == 1) {
+            fmpz_set(&fac->c, G->coeffs);
+            return;
+        }
+
+        fmpz_poly_init(g);
+
+        if (lenG == 2) {
+            fmpz_poly_content(&fac->c, G);
+            if (fmpz_sgn(fmpz_poly_lead(G)) < 0)
+                fmpz_neg(&fac->c, &fac->c);
+            fmpz_poly_scalar_divexact_fmpz(g, G, &fac->c);
+            fmpz_poly_factor_insert(fac, g, 1);
+        } else {
+            slong j, k;
+            fmpz_poly_factor_t sq_fr_fac;
+
+            /* Does a presearch for a factor of form x^k */
+            for (k = 0; fmpz_is_zero(G->coeffs + k); k++)
+                ;
+
+            if (k != 0) {
+                fmpz_poly_t t;
+
+                fmpz_poly_init(t);
+                fmpz_poly_set_coeff_ui(t, 1, 1);
+                fmpz_poly_factor_insert(fac, t, k);
+                fmpz_poly_clear(t);
+            }
+
+            fmpz_poly_shift_right(g, G, k);
+
+            /* Could make other tests for x-1 or simple things
+               maybe take advantage of the composition algorithm */
+            fmpz_poly_factor_init(sq_fr_fac);
+            fmpz_poly_factor_squarefree(sq_fr_fac, g);
+
+            fmpz_set(&fac->c, &sq_fr_fac->c);
+
+            /* Factor each square-free part */
+            for (j = 0; j < sq_fr_fac->num; j++)
+                _fmpz_poly_factor_zassenhaus(fac, sq_fr_fac->exp[j],
+                                             sq_fr_fac->p + j, 10);
+
+            fmpz_poly_factor_clear(sq_fr_fac);
+        }
+        fmpz_poly_clear(g);
+    }
+
+    fmpz_poly_factor_wrapper factors() const
+    {
+        fmpz_poly_factor_wrapper r;
+        fmpz_poly_wrapper::fmpz_poly_factor(poly, *r.get_fmpz_poly_factor_t());
```

Comment:
@isuruf, When FLINT bumps its version, We can modify this to  `fmpz_poly_factor()` instead of `fmpz_poly_wrapper::fmpz_poly_factor()` and remove the above static function.